### PR TITLE
Updated the driver and readme files

### PR DESCRIPTION
**Pull request by dwivedianurag**
In this pr i have updated the driver to be used with akamai and cloudflare and other edge workers. I have also updated the readme files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
-  secretAccessKey: "",
-  service: "dynamodb",
-  endpoint: "http://dynamodb.us-east-2.amazonaws.com",
+import  Client  from "mmdynamo";
+
+const client = new Client({
+  agent: "fetch",
+  apiKey: "xxxx",
+  federationUrl: "http://xxx.xxx.macrometa.io",
+  // this is an optional parameter to be passed when you want to use the endpoint as it is
```
Comment (VikasPatel) [inline README.md:36]: put this inline with `absoluetPath:true`.

File: README.md [hunk 2]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
-   const accessKeyId = "bearer " + JWT;
-   */
-
-const client = new DynamoDB({
-  region,
-  endpoint,
-  accessKeyId,
-  secretAccessKey,
+// this is an optional parameter to be passed when you want to use the endpoint as it is
+const absolutePath=true
```
Comment (VikasPatel) [inline README.md:50]: indentation and semicolon;

File: README.md [hunk 3]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
 ```
 
 ### Create Table
 
 ```js
-client
-  .createTable({...},(err,data)=>{
-    console.log(err,data)
-  })
+      const response = await client.createTable()
```
Comment (VikasPatel) [inline README.md:64]: semicolon

File: README.md [hunk 4]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
+      console.log(response)
```
Comment (VikasPatel) [inline README.md:65]: semicolon

File: README.md [hunk 5]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
-
-### Create Global Table
-
-```js
-client
-  .createGlobalTable({...},(err,data)=>{
-    console.log(err,data)
-  })
+    const response = await client.deleteTable({...})
+      console.log(response)
```
Comment (VikasPatel) [inline README.md:104]: for all instances add semicolon

File: dist/client.d.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Connection } from "./connection";
+import { IConnection } from "./util/types/connectionTypes";
+export default class Client extends Connection {
```
Comment (VikasPatel) [inline dist/client.d.ts:3]: add one line break before this.

File: dist/connection.d.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { IConnection, IOptions } from "./util/types/connectionTypes";
+export declare class Connection {
```
Comment (VikasPatel) [inline dist/connection.d.ts:2]: add one line break

File: dist/util/types/connectionTypes.d.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+export interface IOptions {
+    method: string;
+    headers: headersType;
+    body: string;
+}
+export declare type headersType = {
+    Authorization: string;
+    "x-amz-target": string;
+};
+export declare type endpointsType = {
```
Comment (VikasPatel) [inline dist/util/types/connectionTypes.d.ts:16]: one line break after each.

File: rollup.config.js [hunk 9]
```diff
@@ -2,10 +2,10 @@ import pkg from "./package.json";
 import cleanup from "rollup-plugin-cleanup";
 import { nodeResolve } from "@rollup/plugin-node-resolve";
 import json from "@rollup/plugin-json";
-
+import typescript from '@rollup/plugin-typescript';
 export default {
```
Comment (VikasPatel) [inline rollup.config.js:6]: add one line break;

File: README.md (hunk 1 above)
Comment (dwivedianurag) [reply README.md:36]: done

File: README.md (hunk 2 above)
Comment (dwivedianurag) [reply README.md:50]: done

File: README.md (hunk 3 above)
Comment (dwivedianurag) [reply README.md:64]: done

File: README.md (hunk 1 above)
Comment (VikasPatel) [reply README.md:36]: and put space after `:`

File: README.md (hunk 4 above)
Comment (dwivedianurag) [reply README.md:65]: done

File: README.md (hunk 5 above)
Comment (dwivedianurag) [reply README.md:104]: done

File: dist/client.d.ts (hunk 6 above)
Comment (dwivedianurag) [reply dist/client.d.ts:3]: @VikasPatel these are declaration files created via script do you still want me to add the line break

**Comment by VikasPatel** [comment]
indentation doesn't look correct. check your code.
Screenshot 2021-01-07 at 2 03 20 PM

File: rollup.config.js (hunk 9 above)
Comment (dwivedianurag) [reply rollup.config.js:6]: done

File: dist/connection.d.ts (hunk 7 above)
Comment (dwivedianurag) [reply dist/connection.d.ts:2]: @VikasPatel these are declaration files created via script do you still want me to add the line break

File: dist/util/types/connectionTypes.d.ts (hunk 8 above)
Comment (dwivedianurag) [reply dist/util/types/connectionTypes.d.ts:16]: @VikasPatel these are declaration files created via script do you still want me to add the line break

**Comment by VikasPatel** [comment]
do we still need this line.

Screenshot 2021-01-07 at 2 05 53 PM

File: README.md [hunk 10]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
-```
 
 ### List Tables
 
 ```js
-client.listTables().then((res) => {
-  return res.json();
-});
+      const response = await client.listTables()
+      console.log(response)
```
Comment (VikasPatel) [inline README.md:76]: correct this indentation for all instances. i have added image please check.

Comment (dwivedianurag) [reply README.md:76]: ok sure checking

File: README.md [hunk 11]
```diff
@@ -27,146 +27,85 @@ https://unpkg.com/mmdynamo@0.2.0/dist/mmdynamo.cjs.js
...
 ```js
-import { DynamoDB } from "mmdynamo";
-
-const client = new DynamoDB({
-  region: "us-east-2",
-  accessKeyId: "",
-  secretAccessKey: "",
-  service: "dynamodb",
-  endpoint: "http://dynamodb.us-east-2.amazonaws.com",
+import  Client  from "mmdynamo";
```
Comment (VikasPatel) [inline README.md:30]: Client has extra space around. please remoce.

**Comment by VikasPatel** [comment]
No difference in this two block. check if we need both. i know it was there earlier but now both are same.

Screenshot 2021-01-07 at 2 09 32 PM

**Comment by dwivedianurag** [comment]
> do we still need this line.
> 
> Screenshot 2021-01-07 at 2 05 53 PM

I left this line as a reference for creating table payload, so if you want we can remove it

**Comment by VikasPatel** [comment]
@dwivedianurag Then you need to change the statement.

File: README.md (hunk 1 above)
Comment (dwivedianurag) [reply README.md:36]: done

File: README.md (hunk 10 above)
Comment (dwivedianurag) [reply README.md:76]: done

File: README.md (hunk 11 above)
Comment (dwivedianurag) [reply README.md:30]: done

**Comment by dwivedianurag** [comment]
@VikasPatel updated the statement

**Review by VikasPatel — APPROVED** [review]

**Pull request merged**